org.eclipse.jdt.core/codeassist/org/eclipse/jdt/internal/codeassist/complete/CompletionOnAnnotationMemberValuePair.java

```diff
@@ -33,11 +33,13 @@ public CompletionOnAnnotationMemberValuePair(TypeReference type, int sourceStart
 	public TypeBinding resolveType(BlockScope scope) {
 		super.resolveType(scope);
 
+		CompletionNodeFound exception;
 		if (this.resolvedType == null || !this.resolvedType.isValidBinding()) {
-			throw new CompletionNodeFound();
+			exception = new CompletionNodeFound();
 		} else {
-			throw new CompletionNodeFound(this.completedMemberValuePair, scope);
+			exception = new CompletionNodeFound(this.completedMemberValuePair, scope);
 		}
+		return exception.throwOrDeferAndReturn(() -> this.resolvedType);
```

Comment:
What explains the argument difference between here and COSNR ??

Reply:
Here we unconditionally invoke `super.resolveType()` so `resolvedType` is already assigned if possible. In COSNR that is not the case so I had to "invent" a way to return smth useful in case we defer throwing. But on second look I think that "invention" is not very useful, , and returning a `ProblemReferenceBinding` seems more honest (plus it passes tests, too).